ccan/ccan/graphql/graphql.c

```diff
@@ -0,0 +1,1031 @@
+/* MIT (BSD) license - see LICENSE file for details */
+#include "graphql.h"
+
+#include "ccan/tal/str/str.h"
+#include "ccan/utf8/utf8.h"
+
+
+// GraphQL character classes
+
+#define SOURCE_CHAR(c) ((c) == 9 || (c) == 10 || (c) == 13 || ((c) >= 32 && (c) <= 65535))
```

Comment:
65535?  This is a very strange thing to ask of a char?

Reply:
Ah, I see you have 0x2026.  Which really needs a #define, I think.  And this test still doesn't make sense AFAICT?

Reply:
The number 65535 comes from the spec, and yes, it's indeed a bit strange. The spec does not assume a particular encoding of the input text (encoding is implementation-specific) and it restricts the meaningful input characters to a subset of the 7-bit ASCII range anyway, so in theory, this number can only come into play in certain comments that are allowed to contain arbitrary code points according to the implementation's chosen encoding. But since we treat input as chars (i.e. UTF-8) without validating multibyte code points, this ends up not being used. And since the comments are stripped anyway, there's really not much point in validating multibyte code points in the input. The only real benefit might be for more accurately tracking "character position" for error reporting, but does one cause an error in a comment, anyway? So in the end, yeah, it's kinda just a useless strange thing which is there only because the macro is intended to translate what the spec says, more or less verbatim (an intent which I've now added to the comments).

As for 0x2026, this constant is now defined in token_type_enum as PUNCT_SPREAD to signify the GraphQL spread operator (...). (The constant itself was "arbitrarily" chosen to be the Unicode code point for the ellipsis character, something out of the ASCII range.)